src/runtime-rs/crates/hypervisor/src/qemu/network.rs

```diff
@@ -0,0 +1,372 @@
+    // "virtio-net-pci"
+    #[default]
+    VirtioNetPci,
+    // VirtioNetCCW is the virt-io ccw networking device driver.
+    // "virtio-net-ccw"
+    VirtioNetCcw,
+    //VhostUserNet represents a net vhostuser device type.
+    // "virtio-net"
+    VhostUserNet,
+    // VfioPCI is the vfio driver with PCI transport.
+    // "vfio-pci"
+    VfioPci,
+    // VfioCCW is the vfio driver with CCW transport.
+    // "vfio-ccw"
+    VfioCcw,
+    // VfioAP is the vfio driver with AP transport.
+    // "vfio-ap"
+    VfioAp,
+}
+
+impl ToString for NetDeviceWithTransport {
+    fn to_string(&self) -> String {
+        match self {
+            NetDeviceWithTransport::VirtioNetPci => "virtio-net-pci".to_owned(),
+            NetDeviceWithTransport::VirtioNetCcw => "virtio-net-ccw".to_owned(),
+            NetDeviceWithTransport::VhostUserNet => "virtio-net".to_owned(),
+            NetDeviceWithTransport::VfioPci => "vfio-pci".to_owned(),
+            NetDeviceWithTransport::VfioCcw => "vfio-ccw".to_owned(),
+            NetDeviceWithTransport::VfioAp => "vfio-ap".to_owned(),
+        }
+    }
+}
+
+impl TryFrom<&str> for NetDeviceWithTransport {
+    type Error = anyhow::Error;
+
+    // method to parse a PciPath from a string
+    fn try_from(s: &str) -> Result<Self> {
+        let transport = match s {
+            "virtio-net-pci" => NetDeviceWithTransport::VirtioNetPci,
+            "virtio-net-ccw" => NetDeviceWithTransport::VirtioNetCcw,
+            "virtio-net" => NetDeviceWithTransport::VhostUserNet,
+            "vfio-pci" => NetDeviceWithTransport::VfioPci,
+            "vfio-ccw" => NetDeviceWithTransport::VfioCcw,
+            "vfio-ap" => NetDeviceWithTransport::VfioAp,
+            _ => return Err(anyhow!("unsupported transport")),
+        };
+
+        Ok(transport)
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+enum NetDeviceDriver {
+    #[default]
+    VirtioNet,
+    Vhostuser,
+    Vfio,
+}
+
+impl ToString for NetDeviceDriver {
+    fn to_string(&self) -> String {
+        match self {
+            NetDeviceDriver::VirtioNet => "virtio-net".to_owned(),
+            NetDeviceDriver::Vhostuser => "vhost-user".to_owned(),
+            NetDeviceDriver::Vfio => "vfio".to_owned(),
+        }
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+enum NetDeviceType {
+    /// Tap is a tap networking device type.
+    /// TAP NetDeviceType = "tap"
+    #[default]
+    Tap,
+    /// MacTap is a macvtap networking device type.
+    /// MACVTAP NetDeviceType = "macvtap"
+    MacvTap,
+    /// IpTap is a ipvtap virtual networking device type.
+    /// IPVTAP NetDeviceType = "ipvtap"
+    IpvTap,
+    /// VethTap is a veth-tap virtual networking device type.
+    /// VETHTAP NetDeviceType = "vethtap"
+    VethTap,
+    /// VhuNet is a vhost-user port (socket)
+    /// VhuNet NetDeviceType = "vhostuser"
+    VhuNet,
+    /// Physical is a direct assigned PCI device or PCI VF
+    /// Physical NetDeviceType = "VFIO"
+    Physical,
```

Comment:
When assigning a device from the host using vfio, all network traffic goes directly to the network the device is plugged into, with no `-netdev` on the command line. You likely don't need this.